.github/workflows/release.yml

```diff
@@ -1,63 +1,50 @@
-name: Deploy release candidate
+name: Release
 on:
     push:
         branches:
             - main
     pull_request: # Runs on pull requests to any branch
 env:
-    IS_RELEASE_CANDIDATE: ${{ contains(github.head_ref, 'release--') }}
-    IS_FEATURE: ${{ !contains(github.head_ref, 'release--') }}
     DFX_VERSION: 0.21.0
     NODE_VERSION: 20
 jobs:
-    get-tests:
+    determine-should-release:
         if: ${{ contains(github.head_ref, 'release--') }}
+        name: Determine if this branch should release
+        runs-on: ubuntu-latest
+        outputs:
+            should_release: ${{ steps.determine_should_release.outputs.should_release }}
+        steps:
+            - uses: actions/checkout@v4
+
+            - id: determine_should_release
+              uses: ./.github/actions/should_release
+
+    get-tests:
```

Comment:
Rename test to testInfo...get-test-infos...etc

Reply:
done